Makefile

```diff
@@ -28,88 +28,65 @@ iteration=0
-	docker cp dockerInstance:/tmp/build/output/ ${PWD}/build/rpm
-	docker system prune -a -f
+	docker build -t apache/apisix-dashboard:$(version) --build-arg checkout_v=$(checkout) \
+		-f ./dockerfiles/Dockerfile.dashboard.rpm .
 
 ### build rpm for apisix dashboard:
 .PHONY: package-dashboard-rpm
 package-dashboard-rpm:
-	fpm -f -s dir -t rpm \
-		-n apisix-dashboard \
-		-a `uname -i` \
-		-v $(version) \
-		--iteration $(iteration) \
-		--description 'Apache APISIX Dashboard is designed to make it as easy as possible for users to operate Apache APISIX through a frontend interface.'  \
-		--license "ASL 2.0" \
-		-C ${PWD}/build/rpm/output/apisix/dashboard/ \
-		-p ${PWD}/output/ \
-		--url 'https://github.com/apache/apisix-dashboard'
-
-	rm -rf ${PWD}/build
+	docker build -t apache/apisix-dashboard-packaged:$(version) \
+	--build-arg VERSION=$(version) \
+	--build-arg ITERATION=$(iteration) \
+	--build-arg PACKAGE_VERSION=$(version) \
+	-f ./dockerfiles/Dockerfile.package.apisix-dashboard .
+	docker run -d --name output --net="host" apache/apisix-dashboard-packaged:$(version)
+	docker cp output:/output ${PWD}/output
+	docker stop output
+	docker system prune -a -f
 
 ### build apisix-openresty:
 .PHONY: build-apisix-openresty-rpm
 build-apisix-openresty-rpm:
-	mkdir -p ${PWD}/build/rpm
 	docker build -t apache/apisix-openresty:$(version) --build-arg version=$(version) \
 		-f ./dockerfiles/Dockerfile.apisix-openresty.rpm .
-	docker run -d --name dockerInstance --net="host" apache/apisix-openresty:$(version)
-	docker cp dockerInstance:/usr/local/openresty/ ${PWD}/build/rpm
-	docker system prune -a -f
 
 ### build rpm for apisix-openresty:
 .PHONY: package-apisix-openresty-rpm
 package-apisix-openresty-rpm:
-	fpm -f -s dir -t rpm \
-		-n apisix-openresty \
-		-a `uname -i` \
-		-v $(version) \
-		--iteration $(iteration) \
-		-x openresty/zlib \
-		-x openresty/openssl111 \
-		-x openresty/pcre \
-		-d 'openresty-zlib >= 1.2.11-3' \
-		-d 'openresty-openssl111 >= 1.1.1h-1' \
-		-d 'openresty-pcre >= 8.44-1' \
-		--post-install ${PWD}/post-install-apisix-openresty.sh \
-		--description "APISIX's OpenResty distribution." \
-		--license "ASL 2.0" \
-		-C ${PWD}/build/rpm \
-		-p ${PWD}/output/ \
-		--url 'http://apisix.apache.org/' \
-		--conflicts openresty \
-		--config-files usr/lib/systemd/system/openresty.service \
-		--prefix=/usr/local
-	rm -rf ${PWD}/build
+	docker build -t apache/apisix-openresty-packaged:$(version) \
+	--build-arg VERSION=$(version) \
+	--build-arg ITERATION=$(iteration) \
+	--build-arg PACKAGE_VERSION=$(version) \
+	-f ./dockerfiles/Dockerfile.package.apisix-openresty .
+	docker run -d --name output --net="host" apache/apisix-openresty-packaged:$(version)
+	docker cp output:/output ${PWD}
+	docker stop output
+	docker system prune -a -f
+
+
+### build fpm for packaging:
+.PHONY: build-fpm
+build-fpm:
+	docker build -t api7/fpm - < ./dockerfiles/Dockerfile.fpm
```

Comment:
Why not use `-f`?

Reply:
if use `-f`, we need to pass the `context` which include files that are not necessary for building an image. This will results in a larger build context and larger image size.